complex/ft_test.c

```diff
@@ -310,6 +312,25 @@ void ft_next_iov_cnt(struct ft_xcontrol *ctrl, size_t max_iov_cnt)
 		ctrl->iov_iter = 0;
 }
 
+int ft_next_ctx(struct ft_xcontrol *ctrl, uint64_t comp)
+{
+	int ret;
+
+	ctrl->curr_ctx++;
+	if (ctrl->curr_ctx < ctrl->max_credits)
+		return 0;
+
+	while (ctrl->credits < ctrl->max_credits && comp == FI_SEND) {
+		ret = ft_comp_tx(FT_COMP_TO);
+		if (ret)
+			return ret;
+	}
```

Comment:
Do we actually need the above checks? ft_send_msg and ft_recv_msg already check for (!ft_tx_ctrl.credits) and (ft_rx_ctrl.credits > (ft_rx_ctrl.max_credits >> 1)) respectively. That could take care of ctrl->curr_ctx not exceeding ctrl->max_credits. Then we would just have to increment ctrl->curr_ctx % ctrl->max_credits.